covalent/cloud_resource_manager/core.py

```diff
@@ -0,0 +1,309 @@
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+
+import importlib
+import os
+import shutil
+import subprocess
+from configparser import ConfigParser
+from pathlib import Path
+from types import ModuleType
+from typing import Any, Dict, Optional
+
+from covalent._shared_files.config import set_config
+from covalent._shared_files.exceptions import CommandNotFoundError
+from covalent.executor import _executor_manager
+
+
+def get_executor_module(executor_name: str) -> ModuleType:
+    """
+    Get the executor module from the executor name
+
+    Args:
+        executor_name: Name of the executor
+
+    Returns:
+        The executor module
+
+    """
+    return importlib.import_module(
+        _executor_manager.executor_plugins_map[executor_name].__module__
+    )
+
+
+def get_converted_value(value: str) -> Any:
+    """
+    Convert the value to the appropriate type
+
+    Args:
+        value: Value to be converted
+
+    Returns:
+        Converted value with appropriate type
+
+    """
+    if value.lower() == "true":
+        return True
+    elif value.lower() == "false":
+        return False
+    elif value.lower() == "null":
+        return None
+    elif value.isdigit():
+        return int(value)
+    elif value.replace(".", "", 1).isdigit():
+        return float(value)
+    else:
+        return value
+
+
+def validate_options(executor_options: Dict[str, str], executor_name: str) -> None:
+    """
+    Validate the options passed to the CRM
+
+    Args:
+        executor_options: Options passed to the CRM
+        executor_name: Name of the executor
+
+    Raises:
+        pydantic.ValidationError: If the options are invalid
+
+    """
+    # Importing validation classes from the executor module
+    module = get_executor_module(executor_name)
+    ExecutorPluginDefaults = getattr(module, "ExecutorPluginDefaults")
+    ExecutorInfraDefaults = getattr(module, "ExecutorInfraDefaults")
+
+    # Validating the passed options:
+
+    plugin_attrs = list(ExecutorPluginDefaults.schema()["properties"].keys())
+    infra_attrs = list(ExecutorInfraDefaults.schema()["properties"].keys())
+
+    plugin_params = {k: v for k, v in executor_options.items() if k in plugin_attrs}
+    infra_params = {k: v for k, v in executor_options.items() if k in infra_attrs}
+
+    # Validate options
+    ExecutorPluginDefaults(**plugin_params)
+    ExecutorInfraDefaults(**infra_params)
+
+
+class CloudResourceManager:
+    """
+    Base cloud resource manager class
+    """
+
+    def __init__(
+        self,
+        executor_name: str,
+        executor_module_path: str,
+        options: Optional[Dict[str, str]] = None,
+    ):
+        self.executor_name = executor_name
+        self.executor_tf_path = str(
```

Comment:
@kessler-frost let's update this path to be outside of side packages, ideally in a place not affected by soft purges

Reply:
I've updated the terraform state path so that the state files will not be in site packages but somewhere where they don't get affected by purges. This particular path however I don't think can be changed without significant changes in executor plugin repos.

Reply:
Got it ✔️